test/integration/rmi_out_test.py

```diff
@@ -43,27 +51,146 @@ def test_deprish_out(rmi_out, request):
+            'utilities'
+        ),
+        (
+            "grab_deprish",
+            "grab_deprish_to_eia",
+            'plant_balance_w_common',
+            'plants',
+        ),
+        (
+            "grab_deprish_to_eia",
+            "grab_deprish_to_ferc1",
+            'plant_balance_w_common',
+            'utilities',
+        ),
+        (
+            "grab_deprish_to_eia",
+            "grab_deprish_to_ferc1",
+            'plant_balance_w_common',
+            'plants',
+        ),
+        (
+            "grab_deprish",
+            "grab_deprish_to_ferc1",
+            'plant_balance_w_common',
+            'utilities'
+        ),
+        (
+            "grab_deprish",
+            "grab_deprish_to_ferc1",
+            'plant_balance_w_common',
+            'plants',
+        ),
+        (
+            "grab_ferc1_to_eia",
+            "grab_deprish_to_ferc1",
+            'capex_total',
+            'utilities',
+        ),
+        (
+            "grab_ferc1_to_eia",
+            "grab_deprish_to_ferc1",
+            'capex_total',
+            'plants',
+        ),
+    ]
+)
+def test_data_col_at_two_stages(
+    rmi_out,
+    df1_name: str,
+    df2_name: str,
+    data_col: str,
+    by_name: Literal['plants', 'utilities'],
+):
+    """
+    Test the consistency of a data column at two stages.
+
+    The data that is processed in this repo goes along multiple stages of its
+    journey. There are some quantities that are expected to be invariant when
+    aggregated at the plant level, and the utility level -- that the quantity
+    might slosh around between generators in a plant, or between plants within
+    a utility, but that when aggregated to the plant or utility level, it
+    should be the same at every step in the processing.
+
+    Ideally, the inputs
+    """
+    if by_name == 'plants':
+        by = PK_PLANTS
+    elif by_name == 'utilities':
+        by = PK_UTILS
+
+    test_stages = pudl_rmi.connect_deprish_to_ferc1.gb_test(
+        df1=rmi_out.__getattribute__(df1_name)().pipe(_add_data_source),
+        df2=rmi_out.__getattribute__(df2_name)().pipe(_add_data_source),
+        data_col=data_col,
+        by=by
+
+    )
+    # we add the off_rate.notnll() mask in here bc there are a ton of unconnected
+    # data in these tables (eg. depreciation records that don't have a EIA connection)
+    # these bbs will not get effectively propegated through each of these stages that
+    # require a match to EIA.
+    # there are enough of them that we want to generally ignore them
+    fails = test_stages[~test_stages.match & test_stages.off_rate.notnull()]
+
+    logger.info(
+        f"Failures for {data_col} by {by_name} btwn {df1_name.replace('grab_', '')} "
+        f"and {df2_name.replace('grab_', '')}: {len(fails)}"
+    )
+
+    fail_path = (
+        PATH_FAIL /
+        f"fail_{data_col}_{by_name}_{df1_name.replace('grab_', '')}_vs_"
+        f"{df2_name.replace('grab_', '')}.csv"
+    )
```

Comment:
These string / path constructions / replacements are pretty hard to read and get used more than once. Could you construct them at the beginning and give them a readable variable name?

Reply:
yeah i was wondering about this. I've saved the simple names as variables for now...

but I have to do this because the cooridnate methods are all `grab_{table_name}`.. I could either remove the "grab_" at the source.. or pass in the simple names as arguments into this test and use `rmi_out.__getattribute__(f"grab_{df1_name}")`?

do either of other options sound better to you?